cassandra/src/main/scala/geotrellis/spark/io/cassandra/CassandraCollectionReader.scala

```diff
@@ -58,18 +57,20 @@ object CassandraCollectionReader {
     else
       queryKeyBounds.flatMap(decomposeBounds)
 
-    val query = QueryBuilder.select("value")
-      .from(keyspace, table)
-      .where(eqs("key", QueryBuilder.bindMarker()))
-      .and(eqs("name", layerId.name))
-      .and(eqs("zoom", layerId.zoom))
-      .toString
+    val query = indexStrategy.queryValueStatement(
+      instance.cassandraConfig.indexStrategy,
+      keyspace, table, layerId.name, layerId.zoom
+    )
 
     instance.withSessionDo { session =>
-      val statement = session.prepare(query)
+      val statement = indexStrategy.prepareQuery(query)(session)
```

Comment:
Made the evaluation of `session` lazy since you might want to get clever with a function that evaluates to a `Session` only when in the proper scope for it to be bound to a particular Executor in Spark. 

Maybe I'm overthinking this... but _not_ evaluating my Cassandra sessions lazily has been something that's bitten me before in other Spark code.